docs/tutorials/05_DataAssociation-Clutter.py

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+"""
+5 - Data Association - clutter tutorial
+=======================================
+"""
+
+# %%
+# Tracking a single target in the presence of clutter and missed detections
+# -------------------------------------------------------------------------
+#
+# Tracking is frequently complicated by the presence of detections which are not
+# associated with the target of interest. These may range from sensor-generated noise to
+# returns off intervening physical objects, or environmental effects. We refer to these collectively
+# as *clutter* if they have only nuisance value and need to be filtered out.
+#
+# In this tutorial we introduce the use of data association algorithms in Stone Soup and demonstrate
+# how they can mitigate confusion due to clutter. To begin with we use a **nearest neighbour**
+# method, which is simple and associates a prediction and a detection based only on their proximity
+# as quantified by a particular metric.
+
+# %%
+# As in previous tutorials, we start with a target moving linearly in the 2D cartesian plane.
+import numpy as np
+from scipy.stats import uniform
+from datetime import datetime
+from datetime import timedelta
+
+from stonesoup.models.transition.linear import CombinedLinearGaussianTransitionModel, \
+                                               ConstantVelocity
+from stonesoup.types.groundtruth import GroundTruthPath, GroundTruthState
+
+start_time = datetime.now()
+transition_model = CombinedLinearGaussianTransitionModel([ConstantVelocity(0.005),
+                                                          ConstantVelocity(0.005)])
+truth = GroundTruthPath([GroundTruthState([0, 1, 0, 1], timestamp=start_time)])
+for k in range(1, 21):
+    truth.append(GroundTruthState(
+        transition_model.function(truth[k-1], noise=True, time_interval=timedelta(seconds=1)),
+        timestamp=start_time+timedelta(seconds=k)))
+
+# %%
+# Simulating clutter
+# ^^^^^^^^^^^^^^^^^^
+# Next, generate some measurements and add in some clutter
+# at each time-step. We use the :class:`~.TrueDetection` and :class:`~.Clutter` subclasses of
+# :class:`~.Detection` to help with discerning data types later in plotting. We'll introduce the
+# possibility that, at any time-step, our sensor receives no detection from the target (i.e.
+# :math:`p_d < 1`). A fixed number of clutter points are generated uniformly distributed across
+# a region around the target.
```

Comment:
Could be a bit more explicit about how uniform the clutter actually is (i.e. +-20 around the target).

Reply:
I only had a brief look but I think that, in this instance, clutter is generated by sampling uniformly from a rectangular area around the true position of the target a specified number of times. What the sensor is comes later.

Perhaps whoever wrote originally can confirm? @sdhiscocks @oharrald-Dstl ?